reascripts/ReaSpeech/source/TranscriptExporter.lua

```diff
@@ -28,294 +27,443 @@ function TranscriptExporter:init()
-      self:debug('no formats to set for default')
-      return
-    end
-
-    self.selected_format_key = self.formatters[1].key
-  end
-
-  local index = self.format_map[self.selected_format_key]
-
-  return self.formatters[index]
+    local index = self.format_map[self:selected_key()]
+    return self.formatters[index]
 end
 
 function TranscriptExporterFormats:render_format_options(options)
-  Trap(function()
-    local format = self:selected_format()
+    Trap(function()
+        local format = self:selected_format()
 
-    if format then
-      format.option_renderer(options)
-    end
-  end)
+        if format then
+            format.option_renderer(options)
+        end
+    end)
 end
 
 TranscriptExportFormat = Polo {
-  OPTIONS_NOOP = function(_options) end,
-
-  new = function (key, extension, option_renderer, writer_f)
-    return {
-      key = key,
-      extension = extension,
-      option_renderer = option_renderer,
-      writer = writer_f,
-    }
-  end,
+    OPTIONS_NOOP = function(_options)
+    end,
+
+    new = function(key, extension, option_renderer, writer_f)
+        return {
+            key = key,
+            extension = extension,
+            option_renderer = option_renderer,
+            writer = writer_f
+        }
+    end
 }
 
 function TranscriptExportFormat:file_selector_spec()
-  local selector_spec = '%s files (*.%s)\0*.%s\0All files (*.*)\0*.*\0\0'
-  return selector_spec:format(self.key, self.extension, self.extension)
+    local selector_spec = '%s files (*.%s)\0*.%s\0All files (*.*)\0*.*\0\0'
+    return selector_spec:format(self.key, self.extension, self.extension)
 end
 
 function TranscriptExportFormat.exporter_json()
-  return TranscriptExportFormat.new(
-    'JSON', 'json',
-    TranscriptExportFormat.options_json,
-    TranscriptExportFormat.writer_json
-  )
+    return TranscriptExportFormat.new('JSON', 'json', TranscriptExportFormat.options_json,
+        TranscriptExportFormat.writer_json)
 end
 
 function TranscriptExportFormat.options_json(options)
-  local rv, value = ImGui.Checkbox(ctx, 'One Object per Transcript Segment', options.object_per_segment)
-  if rv then
-    options.object_per_segment = value
-  end
+    if not options.json_widgets then
+        local storage = Storage.ExtState.make {
+            section = 'ReaSpeech.Export.JSON',
+            persist = true
+        }
+
+        options.json_widgets = {
+            object_per_segment = ReaSpeechCheckbox.new {
+                state = storage:boolean('object_per_segment', false),
+                label_long = 'One Object per Transcript Segment',  -- Changed from label to label_long
+                help_text = [[Each transcript segment is exported a separate JSON object.]]
+            }
+        }
+    end
+
+    Trap(function()
+        options.json_widgets.object_per_segment:render()
+        options.object_per_segment = options.json_widgets.object_per_segment:value()
+    end)
```

Comment:
i think maybe it might be best to just call `render` instead of also setting the value by calling the widget `value` with each render cycle. maybe we create another group of functions that represent each of these "`if not options.*_widgets`" blocks:

```
function TranscriptExportFormat.widgets_json(options)
  if options.json_options then
    return options.json_options -- already initialized
  end

  local storage = Storage.ExtState.make {...}

  options.json_options = {
    object_per_segment = ReaSpeechCheckbox.new {...},
  }

  return options.json_options
end

function TranscriptExportFormat.options_json(options)
  Trap(function()
    local widgets = TranscriptExportFormat.widgets_json(options)
    widgets.object_per_segment:render()
  end)
end

function TranscriptExportFormat.writer_json(transcript, output_file, options)
  Trap(function()
    local widgets = TranscriptExportFormat.widgets_json(options)

    if widgets.object_per_segment:value() then
      ...
    else
      ...
    end
  end)
end
```

also kinda nice benefit to keep the persistence/widget definitions separate from the rendering stuff. this all gets super noisy so it's a challenge to keep it understandable (clearly, right? lol).